App.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import {
+  Platform,
+  StatusBar,
+  StyleSheet,
+  View,
+  ActivityIndicator
+} from 'react-native';
+import { AppLoading } from 'expo';
+import * as Font from 'expo-font';
+import { Asset } from 'expo-asset';
+import { Ionicons } from '@expo/vector-icons';
+import ApiKeys from './constants/ApiKeys';
+import * as firebase from 'firebase';
+
+import { Provider } from 'react-redux';
+import { PersistGate } from 'redux-persist/integration/react';
+import { store, persistor } from './redux-persist/store';
+
+import AppNavigator from './navigation/AppNavigator';
+// import MainTabNavigator from "./navigation/MainTabNavigator";
+
+interface State {
+  status: 'loading' | 'authenticated' | 'not_authenticated';
+}
```

Comment:
This is Typescript ✨ 

Here I just define an interface (~ a type) which contain one attribute `status` which can be either `loading` | `authenticated` | `not_authenticated`

The magic is that you'll have the auto-completion when using it and TS will warn you if you make a typo 🤗